nixops/known_hosts.py

```diff
@@ -66,4 +66,5 @@ def update(prev_address: str, new_address: str, public_host_key: str) -> None:
     # FIXME: this rewrites known_hosts twice.
     if prev_address != new_address:
         remove(prev_address, public_host_key)
-    add(new_address, public_host_key)
+    if new_address is not None:
```

Comment:
shouldn't remove also be conditioned to this check?

Reply:
Possibly.  I don't know a good way to test changes do this code, so I was not comfortable going beyond a break-fix.

My guess is that none of these were meant to be called with empty IP addresses, and that the Virtualbox plugin is broken somewhere.